js/ui/components/oldClubhouse.js

```diff
@@ -0,0 +1,339 @@
+        this._reposition();
+    }
+
+    setSuggestOpen(suggestOpen) {
+        if (this._openButton.visible)
+            this._openButton.setHighlighted(suggestOpen);
+    }
+
+    setVisible(visible) {
+        this._visible = visible;
+        this._updateVisibility();
+    }
+
+    destroy() {
+        if (this._updateVisibilityId)
+            getClubhouseWindowTracker().disconnect(this._updateVisibilityId);
+        this._closeButton.destroy();
+        this._openButton.destroy();
+    }
+});
+
+var Component = GObject.registerClass({
+}, class OldClubhouseComponent extends Clubhouse.Component {
+    _init() {
+        // Enable hack-mode if we've the old clubhouse and the hack-mode is
+        // disabled because this should be activated to make it work, so hack
+        // mode will be enabled by default for hack 1.0 users
+        const activated = global.settings.get_boolean('hack-mode-enabled');
+        if (!activated && this.getClubhouseApp())
+            global.settings.set_boolean('hack-mode-enabled', true);
+
+        super._init(CLUBHOUSE_IFACE, CLUBHOUSE_ID, CLUBHOUSE_DBUS_OBJ_PATH);
+        this._clubhouseButtonManager = null;
+
+        let keybindingSettings = new Gio.Settings({ schema: SHELL_KEYBINDINGS_SCHEMA });
+        Main.wm.addKeybinding('toggle-clubhouse',
+                              keybindingSettings,
+                              Meta.KeyBindingFlags.NONE,
+                              Shell.ActionMode.NORMAL |
+                              Shell.ActionMode.OVERVIEW,
+                              this._toggle.bind(this));
+
+        this._clubhouseButtonManager = new ClubhouseButtonManager();
+        this._clubhouseButtonManager.connect('open-clubhouse', () => {
+            this.show(global.get_current_time());
+        });
+        this._clubhouseButtonManager.connect('close-clubhouse', () => {
+            this.hide(global.get_current_time());
+        });
+    }
+
+    _toggle() {
+        if (this._enabled)
+            super.toggle();
+    }
+
+    get _useClubhouse() {
+        return this._imageUsesClubhouse() && !!this.getClubhouseApp();
+    }
+
+    enable() {
+        super.enable();
+        const newHack = !Clubhouse.getClubhouseApp();
+        this._clubhouseButtonManager.displayOpenButton(newHack);
+    }
+
+    // Override this because we need to make sure we only call a property on the proxy once
+    // it's initialized.
+    _onProxyConstructed(object, res) {
+        super._onProxyConstructed(object, res);
+
+        // Make sure the proxy didn't fail to initialize
+        if (this.proxy.SuggestingOpen !== undefined && this._clubhouseButtonManager)
+            this._clubhouseButtonManager.setSuggestOpen(!!this.proxy.SuggestingOpen);
+    }
+
+    _syncVisibility() {
+        super._syncVisibility();
+        if (this._clubhouseButtonManager)
+            this._clubhouseButtonManager.setVisible(this._enabled);
```

Comment:
In old code I see:
```
        if (this._questBanner)
            this._questBanner.actor.visible = this._enabled;

        if (this._itemBanner)
            this._itemBanner.actor.visible = this._enabled;
```
Is it not necessary now?

Reply:
I will reply myself. It is okay with the new changes, since that code is in the parent class.